branch compatibility README-dev (#939)
o1-labs/snarkyjs: README-dev.md

  npm run test:e2e
  npm run e2e:show-report
  ```

## Branch Compatibility

> SnarkyJS is mostly used to write Mina Smart Contract and needs to be compatible with the latest Berkeley testnet (or soon Mainnet), therefor in order to build SnarkyJS from the [Mina repository](https://github.com/MinaProtocol/mina) and make changes to its core, such as the OCaml-bindings in the [snarkyjs-bindings repository](https://github.com/o1-labs/snarkyjs-bindings), all while maintaining compatibility between all repositories, you should follow a certain pattern.

Reviewer: ```suggestion
SnarkyJS is mostly used to write Mina Smart Contracts and must be compatible with the latest Berkeley Testnet (or soon Mainnet). 

To maintain compatibility between the repositories and build SnarkyJS from the [Mina repository](https://github.com/MinaProtocol/mina), make changes to its core, such as the OCaml-bindings in the [snarkyjs-bindings repository](https://github.com/o1-labs/snarkyjs-bindings), you must follow a certain branch compatibility pattern:
```

Reviewer: @Trivo25  what is "its" in

> make changes to its core

Teach me about what and why we change the core? to do ?

Reviewer: did we want to provide some examples? For example, you are working on a feature in <what branch>, so do a PR there, and then cherry pick to < another branch> or ?

Author: By `its core` I mean the SnarkyJS bindings - so everything that is written in OCaml, in the snarkyjs-bindings repository and not directly in Snarkyjs

Reviewer: oh, this significant fact escaped me. wdyt about describing it that way? maybe linking out to the snarkyjs bindings repo?
